### ISSUE TITLE
Midterm Report Review -zw287

### ISSUE THREAD

**Issue by RobinWu218** [medhavi27/Hot-Spotify-Tracks#8]
Summary
The project group is trying to determine the popularity of songs using various features of these songs. They obtained datasets from Spotify Web API and they contain 13 features. In the midterm report, the group explored various models including SVM and Random Forest regression. They also pointed out their future directions. 

Likes:
1. I like how they did feature transformation. For example, interpreting release year as categorical feature is very interesting and captures the intuition that trend of music varies each year. Popularity, however, does not vary linearly with release year, based on our experiences.
2. The train/test data split scheme is insightful. To simulate what we what to achieve in inference time, training data contains songs that were released before the ones in test set. 
3. Initial explorations of models and analysis are clear and informative. Simple linear models cannot fully capture the underlying distribution. It means more complicated, non-linear models should be employed and further feature engineering is necessary. 

Suggestions:
1. I would suggest that the group try to clean the dataset in the next step. As they mentioned, the dataset contains a lot of songs that have popularity 0. In other words, many data points cannot contribute to learning what features of songs lead to popularity. They can select a few ranges of popularity scores (0 to 0.2, 0.2 to 0.4, etc.) and assign roughly the same number of songs to each range.
2. They can also try feature selection before adding more features. For example, the l1 norm can be used to select useful features. Among the 13 features in the existing dataset, they can explore which factors are most important in determining the popularity of a song.
3. They can try using neural networks and kernel methods. If the dataset is truly non-linear, these are useful methods to capture non-linearity.
